Plugins/GeoModel/src/GeoModelToDetVol.cpp

```diff
@@ -0,0 +1,197 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "Acts/Plugins/GeoModel/GeoModelToDetVol.hpp"
+
+#include "Acts/Detector/GeometryIdGenerator.hpp"
+#include "Acts/Detector/PortalGenerators.hpp"
+#include "Acts/Geometry/CuboidVolumeBounds.hpp"
+#include "Acts/Geometry/CutoutCylinderVolumeBounds.hpp"
+#include "Acts/Geometry/CylinderVolumeBounds.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Geometry/TrapezoidVolumeBounds.hpp"
+#include "Acts/Navigation/InternalNavigation.hpp"
+
+#include <GeoModelKernel/GeoBox.h>
+#include <GeoModelKernel/GeoPcon.h>
+#include <GeoModelKernel/GeoShapeShift.h>
+#include <GeoModelKernel/GeoShapeSubtraction.h>
+#include <GeoModelKernel/GeoShapeUnion.h>
+#include <GeoModelKernel/GeoSimplePolygonBrep.h>
+#include <GeoModelKernel/GeoTrd.h>
+#include <GeoModelKernel/GeoTube.h>
+#include <GeoModelKernel/GeoTubs.h>
+
+namespace Acts {
+namespace GeoModel {
+std::shared_ptr<Experimental::DetectorVolume> convertVolume(
+    const GeometryContext& context, const GeoShape& shape,
+    const std::string& name, const Transform3& transform) {
+  auto portalGenerator = Experimental::defaultPortalAndSubPortalGenerator();
+  if (shape.typeID() == GeoTube::getClassTypeID()) {
+    const GeoTube& tube = dynamic_cast<const GeoTube&>(shape);
```

Comment:
If we do the type check with the `typeID()` here (which seems very good to me), we should do a `static_cast` in the following, since this will be a lot cheaper computationally.

Reply:
I would assume that `GeoModel` is consistent in a way, that if the `typeID` matches, the `static_cast` is save. I haven't checked though.

Actually I think `dynamic_cast` throws `std::bad_cast` when used with references, so in a way this is checked... 
I still would go with static cast here, even though it does not really matter here, but of course this is not really important.

I think it could matter more for the surface conversion, where the dynamic casts are done several thousand times... I want to refactor this at some point I think...

Reply:
ok, I changed from `dyanmic_cast` to `static_cast`